test-engine-core-modules/src/rmodel/module_tests/plugin_test.py

```diff
@@ -0,0 +1,174 @@
+import sys
+from pathlib import Path
+from typing import Tuple, Union
+
+from test_engine_core.interfaces.imodel import IModel
+from test_engine_core.interfaces.iserializer import ISerializer
+from test_engine_core.plugins.enums.model_plugin_type import ModelPluginType
+from test_engine_core.plugins.enums.plugin_type import PluginType
+from test_engine_core.plugins.plugins_manager import PluginManager
+from test_engine_core.utils.time import time_class_method
+
+
+# =====================================================================================
+# NOTE: Do not modify this file unless you know what you are doing.
+# =====================================================================================
+class PluginTest:
+    """
+    The PluginTest class specifies methods in supporting testing for the plugin.
+    """
+
+    def __init__(self, model_path: str, discover_path: Path):
+        # Other variables
+        self._base_path: Path = discover_path
+
+        # Store the input arguments as private vars
+        self._model_path: str = str(self._base_path / model_path)
+
+        # Default for instances
+        self._model_instance: Union[None, IModel] = None
+        self._model_serializer_instance: Union[None, ISerializer] = None
+
+        self._expected_model_algorithm = "rpy2.robjects.vectors.ListVector"
+        self._expected_model_plugin_type = ModelPluginType.R
+
+    @time_class_method
+    def run(self) -> None:
+        """
+        A function to run the plugin test with the provided arguments.
+        """
+        try:
+            error_count = 0
+            error_message = ""
+
+            # Load all the core plugins and the model plugin
+            PluginManager.discover(str(self._base_path))
+
+            # Get the model instance
+            (
+                self._model_instance,
+                self._model_serializer_instance,
+                error_message,
+            ) = PluginManager.get_instance(
+                PluginType.MODEL, **{"filename": self._model_path}
+            )
+
+            # Perform model instance setup
+            is_success, error_messages = self._model_instance.setup()
+            if not is_success:
+                raise RuntimeError(
+                    f"Failed to perform model instance setup: {error_messages}"
+                )
+
+            # Run different tests on the model instance
+            test_methods = [
+                self._validate_metadata,
+                self._validate_plugin_type,
+                self._validate_model_supported,
+                self._
```

Comment:
Hi, this line seems to be causing an error.

Reply:
Ah yes,  I have removed that one now